sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -60,11 +60,12 @@ case class DataSourceResolution(
     case create: CreateTableStatement =>
       // the provider was not a v1 source, convert to a v2 plan
       val CatalogObjectIdentifier(maybeCatalog, identifier) = create.tableName
-      val catalog = maybeCatalog.orElse(defaultCatalog)
-          .getOrElse(throw new AnalysisException(
-            s"No catalog specified for table ${identifier.quoted} and no default catalog is set"))
-          .asTableCatalog
-      convertCreateTable(catalog, identifier, create)
+      maybeCatalog match {
+        case Some(catalog) =>
+          convertCreateTable(catalog.asTableCatalog, identifier, create)
+        case _ =>
+          convertCreateTable(v2SessionCatalog.asTableCatalog, identifier, create)
```

Comment:
Should it be default catalog instead of session catalog here? It would be good to have some comments here.

Reply:
I've updated this: